activesupport/test/cache/stores/redis_cache_store_test.rb

```diff
@@ -283,6 +283,40 @@ def ensure_connected
     end
   end
 
+  class FailureRaisingFromUnavailableClientTest < StoreTest
+    include FailureRaisingBehavior
+
+    private
+      def emulating_unavailability
+        old_client = Redis.send(:remove_const, :Client)
+        Redis.const_set(:Client, UnavailableRedisClient)
+
+        assert_raise Redis::BaseConnectionError do
```

Comment:
I don't think `assert_raise` should be part of `emulating_unavailability`, because it makes it very unclear in the calling test.

When I was reading them I was wondering why no error was raised even thought it was the goal of your PR.

Reply:
Haha I just wrote the exact same comment ^